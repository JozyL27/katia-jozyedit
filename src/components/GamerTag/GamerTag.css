.gamer-tag__container {
    display: flex;
    justify-content: space-between;
    width: 90%;
    margin: 2vh auto;
}

label.gamer-tag__label {
    background: white;
    padding: .7vh 1.5vh;
    font-size: 3vh;
    font-weight: normal;
    height: 4.5vh;
    text-align: left;
    width: 35%;
    border-radius: none;
    color: #000;
    display: inline-block;
}

.gamer-tag__input {
    font-size: 3vh;
    padding: .5vh 0 0 1.5vh;
    height: 4.5vh;
    border: .08rem solid gray;
    width: 60%;
    margin: 0 auto;
    text-align: left;
    border-radius: 1vh;
}