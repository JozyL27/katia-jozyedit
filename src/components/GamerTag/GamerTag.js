import React from 'react'
import './GamerTag.css'

export default function GamerTag(props) {

    let formattedInput = props.name.charAt(0).toUpperCase() + props.name.slice(1);
    return(
      <div className='gamer-tag__container'>
        <label 
          htmlFor={props.name}
          className='gamer-tag__label'>
            {props.name === 'psn' ? props.name.toUpperCase() : formattedInput}: </label>
        <input 
            type='text'
            className='gamer-tag__input'
            name={props.name}
            onChange={e => props.handleChange(e)}
            id={props.name}
            defaultValue={props.defaultVal}
        />
      </div>
    )
}