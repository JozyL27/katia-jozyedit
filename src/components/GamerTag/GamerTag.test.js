import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import GamerTag from './GamerTag'

describe('<GamerTag />', () => {
  const testData = {
    name: 'TestTag'
  }
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <MemoryRouter>
          <GamerTag name={testData.name}/>
      </MemoryRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
            <GamerTag name={testData.name}/>
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})