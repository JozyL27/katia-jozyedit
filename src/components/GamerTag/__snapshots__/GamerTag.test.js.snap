// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GamerTag /> renders the UI as expected 1`] = `
<div
  className="gamer-tag__container"
>
  <label
    className="gamer-tag__label"
    htmlFor="TestTag"
  >
    TestTag
    : 
  </label>
  <input
    className="gamer-tag__input"
    id="TestTag"
    name="TestTag"
    onChange={[Function]}
    type="text"
  />
</div>
`;
